src/bscPlugin/SignatureHelpUtil.ts

```diff
@@ -0,0 +1,155 @@
+import { ParameterInformation, SignatureInformation } from 'vscode-languageserver-protocol';
+import { isClassStatement, isXmlScope, isFunctionStatement, isMethodStatement } from '../astUtils/reflection';
+import { TokenKind } from '../lexer/TokenKind';
+import type { Statement } from '../parser/AstNode';
+import { ParseMode } from '../parser/Parser';
+import type { ClassStatement } from '../parser/Statement';
+import type { SignatureInfoObj } from '../Program';
+import type { XmlScope } from '../XmlScope';
+import type { CallExpressionInfo } from './CallExpressionInfo';
+import { CallExpressionType } from './CallExpressionInfo';
+import type { FileLink } from '../interfaces';
+import { util } from '../util';
+
+export class SignatureHelpUtil {
```

Comment:
Should this really be a class? It looks like three basically independent functions....

Reply:
I actually prefer the util class as implemented here. Our primary `Util.ts` file is also a class, so this is following that same pattern. Using a class helps improves the test experience since standalone functions are harder to mock in unit tests.